pkg/sql/conn_executor_exec.go

```diff
@@ -713,7 +713,21 @@ func (ex *connExecutor) dispatchToExecutionEngine(
 		planner.curPlan.flags.Set(planFlagDistSQLLocal)
 	}
 	ex.sessionTracing.TraceExecStart(ctx, "distributed")
+
 	bytesRead, rowsRead, err := ex.execWithDistSQLEngine(ctx, planner, stmt.AST.StatementType(), res, distributePlan, progAtomic)
+	if res.Err() == nil {
+		log.Infof(ctx, "Hello retry: %v / %d / %d", ex.sessionData.RetryErrorsEnabled, planner.Txn().Epoch(), ex.state.lastEpoch)
+		if ex.sessionData.RetryErrorsEnabled && stmt.AST.StatementTag() != "SET" {
+			if planner.Txn().Epoch() < ex.state.lastEpoch+3 {
+				retryErr := planner.Txn().GenerateForcedRetryableError(
+					ctx, "forced by executor")
```

Comment:
nit: I'd say "injected" by executor since it you might not be sure why it was "forced by executor". In. fact, i'd include the name of the flag in the error too, e.g. "error injected by executor due to <setting_name>".